### ISSUE TITLE
Test suite failing to build with latest version of HSpec

### ISSUE THREAD

**Issue by ocharles** [goldfirere/th-desugar#21]
```
ollie@nixos ~/nixpkgs> nix-build -A haskellngPackages.th-desugar
these derivations will be built:
  /nix/store/xg4jy9bkkngv4jj17722ndq3jvmrf8ak-haskell-th-desugar-1.4.2.drv
building path(s) ‘/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2’
building /nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2
unpacking sources
unpacking source archive /nix/store/aj8pi3w9g2i8k2h2w6bb8rm06s153p5s-th-desugar-1.4.2.tar.gz
source root is th-desugar-1.4.2
patching sources
configuring
Building with /nix/store/v2vmaf6w1x0gg3l3a2g83gr5yq7mgl5h-ghc-7.8.4.
[1 of 1] Compiling Main             ( Setup.hs, /tmp/nix-build-haskell-th-desugar-1.4.2.drv-0/Main.o )
Linking Setup ...
configureFlags: --verbose --prefix=/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2 --libdir=$prefix/lib/$compiler --libsubdir=$pkgid --enable-split-objs --disable-library-profiling --enable-shared --enable-library-vanilla --enable-executable-dynamic --enable-tests --ghc-option=-optl=-Wl,-rpath=/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2/lib/ghc-7.8.4/th-desugar-1.4.2 --extra-include-dirs=/nix/store/zbbza1gd0knsv14k6igkp243iwib90wy-gmp-5.1.3/include --extra-lib-dirs=/nix/store/zbbza1gd0knsv14k6igkp243iwib90wy-gmp-5.1.3/lib --extra-include-dirs=/nix/store/d6m9m2wqaymfiwbx11frdsm1cz7qjkdm-ncurses-5.9/include --extra-lib-dirs=/nix/store/d6m9m2wqaymfiwbx11frdsm1cz7qjkdm-ncurses-5.9/lib --package-db=/tmp/nix-build-haskell-th-desugar-1.4.2.drv-0/package.conf.d
Configuring th-desugar-1.4.2...
Dependency HUnit >=1.2: using HUnit-1.2.5.2
Dependency base ==4.*: using base-4.7.0.2
Dependency containers >=0.5: using containers-0.5.5.1
Dependency hspec >=1.3: using hspec-2.1.2
Dependency mtl >=2.1: using mtl-2.1.3.1
Dependency syb >=0.4: using syb-0.4.4
Dependency template-haskell -any: using template-haskell-2.9.0.0
/nix/store/v2vmaf6w1x0gg3l3a2g83gr5yq7mgl5h-ghc-7.8.4/bin/ghc --info
Using Cabal-1.18.1.5 compiled by ghc-7.8
Using compiler: ghc-7.8.4
Using install prefix:
/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2
Binaries installed in:
/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2/bin
Libraries installed in:
/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2/lib/ghc-7.8.4/th-desugar-1.4.2
Private binaries installed in:
/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2/libexec
Data files installed in:
/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2/share/x86_64-linux-ghc-7.8.4/th-desugar-1.4.2
Documentation installed in:
/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2/share/doc/x86_64-linux-ghc-7.8.4/th-desugar-1.4.2
Configuration files installed in:
/nix/store/w8m0hskip76i3ihjsdi5r5qpj3ng5xkn-haskell-th-desugar-1.4.2/etc
No alex found
Using ar found on system at:
/nix/store/wr5c9d4mjb4c7gqvpv2qzg2z4h4lry17-binutils-2.23.1/bin/ar
No c2hs found
No cpphs found
No ffihugs found
Using gcc version 4.8.4 found on system at:
/nix/store/ng41cx2qbqmr81b4cvxvm9qi80kbasv0-gcc-wrapper-4.8.4/bin/gcc
Using ghc version 7.8.4 found on system at:
/nix/store/v2vmaf6w1x0gg3l3a2g83gr5yq7mgl5h-ghc-7.8.4/bin/ghc
Using ghc-pkg version 7.8.4 found on system at:
/nix/store/v2vmaf6w1x0gg3l3a2g83gr5yq7mgl5h-ghc-7.8.4/bin/ghc-pkg
No greencard found
Using haddock version 2.14.3 found on system at:
/nix/store/v2vmaf6w1x0gg3l3a2g83gr5yq7mgl5h-ghc-7.8.4/bin/haddock
No happy found
No hmake found
Using hpc version 0.67 found on system at:
/nix/store/v2vmaf6w1x0gg3l3a2g83gr5yq7mgl5h-ghc-7.8.4/bin/hpc
Using hsc2hs version 0.67 found on system at:
/nix/store/v2vmaf6w1x0gg3l3a2g83gr5yq7mgl5h-ghc-7.8.4/bin/hsc2hs
Using hscolour version 1.20 found on system at:
/nix/store/4fhjn67lip8j3bxhbqb0nmd6rqds2f3g-hscolour-1.20.3/bin/HsColour
No hugs found
No jhc found
Using ld found on system at:
/nix/store/ng41cx2qbqmr81b4cvxvm9qi80kbasv0-gcc-wrapper-4.8.4/bin/ld
No lhc found
No lhc-pkg found
No nhc98 found
No pkg-config found
Using ranlib found on system at:
/nix/store/wr5c9d4mjb4c7gqvpv2qzg2z4h4lry17-binutils-2.23.1/bin/ranlib
Using strip found on system at:
/nix/store/wr5c9d4mjb4c7gqvpv2qzg2z4h4lry17-binutils-2.23.1/bin/strip
Using tar found on system at:
/nix/store/k9ama576yfrfx1837rg4rrilma8pnwks-gnutar-1.27.1/bin/tar
No uhc found
building
Building th-desugar-1.4.2...
Preprocessing library th-desugar-1.4.2...
[1 of 6] Compiling Language.Haskell.TH.Desugar.Util ( Language/Haskell/TH/Desugar/Util.hs, dist/build/Language/Haskell/TH/Desugar/Util.o )
[2 of 6] Compiling Language.Haskell.TH.Desugar.Core ( Language/Haskell/TH/Desugar/Core.hs, dist/build/Language/Haskell/TH/Desugar/Core.o )
[3 of 6] Compiling Language.Haskell.TH.Desugar.Sweeten ( Language/Haskell/TH/Desugar/Sweeten.hs, dist/build/Language/Haskell/TH/Desugar/Sweeten.o )
[4 of 6] Compiling Language.Haskell.TH.Desugar.Expand ( Language/Haskell/TH/Desugar/Expand.hs, dist/build/Language/Haskell/TH/Desugar/Expand.o )
[5 of 6] Compiling Language.Haskell.TH.Desugar ( Language/Haskell/TH/Desugar.hs, dist/build/Language/Haskell/TH/Desugar.o )
[6 of 6] Compiling Language.Haskell.TH.Desugar.Lift ( Language/Haskell/TH/Desugar/Lift.hs, dist/build/Language/Haskell/TH/Desugar/Lift.o )
In-place registering th-desugar-1.4.2...
Preprocessing test suite 'spec' for th-desugar-1.4.2...
[1 of 9] Compiling Language.Haskell.TH.Desugar.Util ( Language/Haskell/TH/Desugar/Util.hs, dist/build/spec/spec-tmp/Language/Haskell/TH/Desugar/Util.dyn_o )
[2 of 9] Compiling Language.Haskell.TH.Desugar.Core ( Language/Haskell/TH/Desugar/Core.hs, dist/build/spec/spec-tmp/Language/Haskell/TH/Desugar/Core.dyn_o )
[3 of 9] Compiling Language.Haskell.TH.Desugar.Sweeten ( Language/Haskell/TH/Desugar/Sweeten.hs, dist/build/spec/spec-tmp/Language/Haskell/TH/Desugar/Sweeten.dyn_o )
[4 of 9] Compiling Language.Haskell.TH.Desugar.Expand ( Language/Haskell/TH/Desugar/Expand.hs, dist/build/spec/spec-tmp/Language/Haskell/TH/Desugar/Expand.dyn_o )
[5 of 9] Compiling Language.Haskell.TH.Desugar ( Language/Haskell/TH/Desugar.hs, dist/build/spec/spec-tmp/Language/Haskell/TH/Desugar.dyn_o )
[6 of 9] Compiling Test.Splices     ( Test/Splices.hs, dist/build/spec/spec-tmp/Test/Splices.dyn_o )
[7 of 9] Compiling Test.DsDec       ( Test/DsDec.hs, dist/build/spec/spec-tmp/Test/DsDec.dyn_o )
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Loading package syb-0.4.4 ... linking ... done.
Loading package transformers-0.3.0.0 ... linking ... done.
Loading package mtl-2.1.3.1 ... linking ... done.
Loading package array-0.5.0.0 ... linking ... done.
Loading package deepseq-1.3.0.2 ... linking ... done.
Loading package HUnit-1.2.5.2 ... linking ... done.
Loading package old-locale-1.0.0.6 ... linking ... done.
Loading package time-1.4.2 ... linking ... done.
Loading package random-1.1 ... linking ... done.
Loading package containers-0.5.5.1 ... linking ... done.
Loading package pretty-1.1.1.1 ... linking ... done.
Loading package template-haskell ... linking ... done.
Loading package primitive-0.5.4.0 ... linking ... done.
Loading package tf-random-0.5 ... linking ... done.
Loading package QuickCheck-2.7.6 ... linking ... done.
Loading package bytestring-0.10.4.0 ... linking ... done.
Loading package unix-2.7.0.1 ... linking ... done.
Loading package ansi-terminal-0.6.2.1 ... linking ... done.
Loading package stm-2.4.4 ... linking ... done.
Loading package async-2.0.2 ... linking ... done.
Loading package hspec-expectations-0.6.1.1 ... linking ... done.
Loading package quickcheck-io-0.1.1 ... linking ... done.
Loading package setenv-0.1.1.3 ... linking ... done.
Loading package hspec-core-2.1.2 ... linking ... done.
Loading package hspec-discover-2.1.2 ... linking ... done.
Loading package hspec-2.1.2 ... linking ... done.
[8 of 9] Compiling Test.Dec         ( Test/Dec.hs, dist/build/spec/spec-tmp/Test/Dec.dyn_o )
[9 of 9] Compiling Test.Run         ( Test/Run.hs, dist/build/spec/spec-tmp/Test/Run.dyn_o )

Test/Run.hs:124:25:
    Ambiguous occurrence ‘runIO’
    It could refer to either ‘Test.Hspec.runIO’,
                             imported from ‘Test.Hspec’ at Test/Run.hs:19:1-17
                             (and originally defined in ‘hspec-core-2.1.2:Test.Hspec.Core.Spec.Monad’)
                          or ‘Language.Haskell.TH.runIO’,
                             imported from ‘Language.Haskell.TH’ at Test/Run.hs:29:1-26
                             (and originally defined in ‘Language.Haskell.TH.Syntax’)
    In the splice:
      $(do { info <- reify ''Poly;
             dinfo@(DTyConI (DFamilyD TypeFam _name _tvbs (Just resK))
                            (Just [DTySynInstD _name2 (DTySynEqn lhs _rhs)])) <- dsInfo info;
             case (resK, lhs) of {
               (DStarK, [DVarT _]) -> ...
               _ -> ... } })

Test/Run.hs:124:25:
    Couldn't match type ‘hspec-core-2.1.2:Test.Hspec.Core.Spec.Monad.SpecM
                           a0’
                  with ‘Q’
    Expected type: Q ()
      Actual type: hspec-core-2.1.2:Test.Hspec.Core.Spec.Monad.SpecM
                     a0 ()
    In a stmt of a 'do' block:
      Test.Hspec.runIO
      $ do { putStrLn "Failed bug8884 test:";
             putStrLn $ show dinfo }
    In the expression:
      do { Test.Hspec.runIO
           $ do { putStrLn "Failed bug8884 test:";
                  putStrLn $ show dinfo };
           [| False |] }
    In a case alternative:
        _ -> do { Test.Hspec.runIO
                  $ do { putStrLn "Failed bug8884 test:";
                         .... };
                  [| False |] }
builder for ‘/nix/store/xg4jy9bkkngv4jj17722ndq3jvmrf8ak-haskell-th-desugar-1.4.2.drv’ failed with exit code 1
error: build of ‘/nix/store/xg4jy9bkkngv4jj17722ndq3jvmrf8ak-haskell-th-desugar-1.4.2.drv’ failed
```

**Comment by goldfirere** [comment]
Fixed and released as 1.4.2.1.

**Comment by ocharles** [comment]
Thank you! :clap:
